### ISSUE TITLE
Memory leak in indexing error

### ISSUE THREAD

**Issue by MeirShpilraien** [RediSearch/RediSearch#994]
Reproduce on 1.4 branch:
```
In [18]: def Test(): 
    ...:     import redis 
    ...:     r = redis.Redis() 
    ...:     for j in range(10000):  
    ...:        query = ['ft.add', 'idx', 'doc%d'%j, '1.0', 'REPLACE', 'FIELDS']  
    ...:        query += ['f1', ','.join([str(a) for a in range(7000)])]  
    ...:        query += ['f2', ','.join([str(a) for a in range(7000)])]  
    ...:        query += ['f3', ','.join([str(a) for a in range(7000)])]  
    ...:        query += ['f4', ','.join([str(a) for a in range(7000)])]  
    ...:        query += ['f5', ','.join([str(a) for a in range(7000)])]  
    ...:        query += ['f6', ','.join([str(a) for a in range(7000)])]  
    ...:        query += ['f7', ','.join([str(a) for a in range(7000)])] 
    ...:        try: 
    ...:            r.execute_command(*query) 
    ...:        except Exception as e: 
    ...:            print(e) 

In [18]: def start(): 
    ...:     p1 = Process(target=Test, args=()) 
    ...:     p2 = Process(target=Test, args=())  
    ...:     p1.start() 
    ...:     p2.start() 
    ...:     p1.join() 
    ...:     p2.join()
In [18]: start()
```

The tags array (`fdata->tags`) of the documents that failed on the ft.add command is leaking in  this case.

**Comment by MeirShpilraien** [comment]
This issue also might cause master slave inconsistency because on slave all the updates are successful.

**Comment by mnunberg** [comment]
I believe this memory leak is fixed in 1.6

**Comment by K-Jo** [comment]
Is fixed, @MeirShpilraien
